File: pingdart-java/src/main/java/com/pingdart/sdk/services/CallsService.java
package com.pingdart.sdk.services;

import com.google.gson.Gson;
import okhttp3.*;

import java.io.IOException;

public class CallsService {
    private final OkHttpClient http;
    private final String baseUrl;
    private final Gson gson = new Gson();

    public CallsService(OkHttpClient http, String baseUrl) {
        this.http = http;
        this.baseUrl = baseUrl;
    }

    public String listApps() throws IOException {
        Request request = new Request.Builder()
                .url(baseUrl + "/v1/calls/apps")
                .get()
                .build();

        try (Response response = http.newCall(request).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }

    public String createApp(String name, String type) throws IOException {
        String json = gson.toJson(new CreateAppRequest(name, type));
        RequestBody body = RequestBody.create(json, MediaType.get("application/json; charset=utf-8"));

        Request request = new Request.Builder()
                .url(baseUrl + "/v1/calls/apps")
                .post(body)
                .build();

        try (Response response = http.newCall(request).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }

    public String deleteApp(String id) throws IOException {
        Request request = new Request.Builder()
                .url(baseUrl + "/v1/calls/apps/" + id)
                .delete()
                .build();

        try (Response response = http.newCall(request).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }

    private static class CreateAppRequest {
        String name;
        String type;
        CreateAppRequest(String name, String type) {
            this.name = name;
            this.type = type;
        }
    }
}

File: pingdart-php/src/Services/CallsService.php
<?php

namespace PingDart\SDK\Services;

use GuzzleHttp\Client;

class CallsService {
    private $http;

    public function __construct(Client $http) {
        $this->http = $http;
    }

    public function listApps() {
        try {
            $response = $this->http->get('v1/calls/apps');
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to list call apps: " . $e->getMessage());
        }
    }

    public function createApp(string $name, string $type = 'Web') {
        try {
            $response = $this->http->post('v1/calls/apps', [
                'json' => ['name' => $name, 'type' => $type]
            ]);
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to create call app: " . $e->getMessage());
        }
    }

    public function deleteApp($id) {
        try {
            $response = $this->http->delete("v1/calls/apps/{$id}");
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to delete call app: " . $e->getMessage());
        }
    }
}
